velox/functions/sparksql/registration/RegisterArray.cpp

```diff
@@ -109,6 +109,27 @@ inline void registerArrayRemoveFunctions(const std::string& prefix) {
       Varchar>({prefix + "array_remove"});
 }
 
+template <typename T>
+inline void registerArrayUnionFunctions(const std::string& prefix) {
+  registerFunction<ArrayUnionFunction, Array<T>, Array<T>, Array<T>>(
+      {prefix + "array_union"});
```

Comment:
There was an attempt in https://github.com/facebookincubator/velox/pull/10453 to fix the array_union function on nested NULL for Presto, could you confirm the behaviour of Spark?

Reply:
That PR has not been merged. Its behavior is now consistent with Spark. A test case has been added to ensure that related changes in function behavior can be detected.